### PR TITLE
Update buildpack task to drop privilege

### PR DESCRIPTION
**Pull request by ImJasonH**
This is an update of https://github.com/tektoncd/catalog/pull/276

cc @ekcasey 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [y] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [y] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [y] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [y] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [y] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [y] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [y] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [y] mandatory `spec.description` follows the convention

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/buildpacks/0.2/README.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+This build template builds source into a container image using [Cloud Native Buildpacks](https://buildpacks.io). To do that, it uses [builders](https://buildpacks.io/docs/concepts/components/builder/#what-is-a-builder) to run buildpacks against your application.
+
+Cloud Native Buildpacks are pluggable, modular tools that transform application source code into OCI images. They replace Dockerfiles in the app development lifecycle, and enable for swift rebasing of images and modular control over images (through the use of builders), among other benefits. This command uses a builder to construct the image, and pushes it to the registry provided.
+
+See also [`buildpacks-phases`](../buildpacks-phases) for the deconstructed version of this task, which runs each of the [lifecycle phases](https://buildpacks.io/docs/concepts/components/lifecycle/#phases) individually (this task uses the [creator binary](https://github.com/buildpacks/spec/blob/platform/0.3/platform.md#operations), which coordinates and runs all of the phases).
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/task/buildpacks/0.1/buildpacks.yaml
```
Comment (piyush-garg) [inline task/buildpacks/0.2/README.md:12]: nit: 0.2

**Comment by chmouel** [comment]
/lgtm

Looking Good! 🤙🏽

File: task/buildpacks/0.2/buildpacks.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    - name: source
+
+  stepTemplate:
+    env:
+      - name: CNB_PLATFORM_API
+        value: "0.3"
+
+  steps:
+    - name: prepare
+      image: alpine
```
Comment (vinamra28) [inline task/buildpacks/0.2/buildpacks.yaml:68]: Can we have the fully qualified name of the image as it's done in #519

File: task/buildpacks/0.2/buildpacks.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+  stepTemplate:
+    env:
+      - name: CNB_PLATFORM_API
+        value: "0.3"
+
+  steps:
+    - name: prepare
+      # Latest alpine as of Oct 22, 2020
+      image: alpine@sha256:203ee936961c0f491f72ce9d3c3c67d9440cdb1d61b9783cf340baa09308ffc1
```
Comment (vinamra28) [inline task/buildpacks/0.2/buildpacks.yaml:69]: nit: (catlin would not be able to validate this image) :sweat_smile: 
```suggestion
      image: docker.io/alpine@sha256:203ee936961c0f491f72ce9d3c3c67d9440cdb1d61b9783cf340baa09308ffc1
```

**Review by vinamra28 — APPROVED** [review]
/lgtm

File: task/buildpacks/0.2/README.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: example-run
+spec:
+  taskRef:
+    name: buildpacks
+  podTemplate:
+    volumes:
+# Uncomment the lines below to use an existing cache
```
Comment (chmouel) [inline task/buildpacks/0.2/README.md:63]: We could use optional workspace here, isnt it ?

Comment (ImJasonH) [reply task/buildpacks/0.2/README.md:63]: We definitely could.

**Pull request merged**